src/main/java/io/vertx/core/http/impl/Http2ServerRequestImpl.java

```diff
@@ -516,4 +520,25 @@ public HttpServerRequest customFrameHandler(Handler<HttpFrame> handler) {
   public HttpConnection connection() {
     return conn;
   }
+
+  @Override
+  public HttpServerRequest streamPriorityHandler(Handler<StreamPriority> handler) {
+    streamPriorityHandler = handler;
+    return this;
+  }
+
+  @Override
+  void handlePriorityChange(StreamPriority streamPriority) {
+    System.out.println(getClass().getName() + ".handlePriorityChange(" + streamPriority + ")");
+    synchronized (conn) {
+      Handler<StreamPriority> handler = streamPriorityHandler;
+      if (handler != null) {
+        if(streamPriority != null && !streamPriority.equals(getStreamPriority())) {
+          setStreamPriority(streamPriority);
+          handler.handle(streamPriority);      
```

Comment:
read the handler in the synchronized block but call handler outside of the synchronized block